lib/mongo_storage.ex

```diff
@@ -174,6 +198,16 @@ defmodule ApplicationRunner.MongoStorage do
     end
   end
 
+  defp handle_delete_doc(pipe, session_uuid, coll, bson_doc_id) do
+    if session_uuid != nil do
+      pipe
+      |> Mongo.delete_one(coll, %{"_id" => bson_doc_id}, session: Swarm.whereis_name(session_uuid))
```

Comment:
Do you really need to pass the session uuid in the delete_one method ? I don't think this is necessary because you already give the _id which should be enough to reference the doc. Maybe I am missing something, what do you think?

Reply:
It's the transaction uuid which is needed to be hable to rollback the all transaction commands.